testbed/interfaces/node.go

```diff
@@ -0,0 +1,93 @@
+package testbedi
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/ipfs/go-cid"
+	"github.com/multiformats/go-multiaddr"
+)
+
+type NewNodeFunc func(dir string, extras map[string]interface{}) (TestbedNode, error)
+type GetAttrListFunc func() []string
+type GetAttrDescFunc func(attr string) (string, error)
+
+// TestbedNode specifies the interface to a process controlled by iptb
+type TestbedNode interface {
```

Comment:
This interface is getting pretty big, I wonder if we should break it up, and make some of the methods optional (by specialization, or similar)

Reply:
The return types and parameters are a little off, but a rough idea:
https://gist.github.com/frrist/01405d6b36aae85108fad80c2c40d3fb